### DISCUSSION TITLE
템플릿 정렬 -> 좋아요 수를 가져오는 방법을 개선해 봐요.

### DISCUSSION THREAD

**Discussion by zangsu** [woowacourse-teams/2024-code-zap#733] [결정 사항]
템플릿 정렬에서 좋아요 수를 가져오기 위해 항상 JOIN 쿼리가 사용되는 점을 개선해 봐요.

**Comment by zangsu** [comment]
JOIN 을 그대로 사용한다면 중간 테이블에 인덱스를 적절하게 사용해 주는 것이 최선일 것 같음

JOIN 사용을 개선하고 싶으면 "좋아요 수" 자체를 컬럼으로 만들고 관리를 해 주어야 할 것 같음
-> 좋아요 순 정렬 때문에 항상 좋아요 수가 필요한 상황
-> 동시성 문제를 잘 제어해 주어야 하지만, 반대로 우리의 학습이 가능할 수 있음

**Comment by zeus6768** [comment]
# 결정
- Tempalte 테이블에 likesCount 칼럼 추가
- Template 도메인에 likesCount 필드 추가
